test/datasets_utils.py

```diff
@@ -952,3 +952,24 @@ def make_fake_flo_file(h, w, file_name):
     )
     with open(file_name, "wb") as f:
         f.write(content)
+
+
+def random_group(collection, groups):
+    """Randomly put items into groups without overlap.
+    Args:
+        collection: Collection of items to be grouped.
+        groups: Collection of group keys.
+    Returns:
+        Dictionary with ``groups`` as keys. Each value is a list of random items from ``collection`` without overlap
+            to the other values. Each list has at least length ``1``.
```

Comment:
Perhaps a docstring example would be helpful to understand what this really does?

Reply:
I refactored to have a `random_subsets` function rather than `random_groups`. The docstring now also includes a usage example. LMK if it is sufficient.